### ISSUE TITLE
How to replicate Global Menu like Macify

### ISSUE THREAD

**Issue by RedBearAK** [Jonchun/macify-linux#40]
How exactly does Macify accomplish the global menu? What files should I look at to be able to replicate this style of global menu on other systems, possibly even other desktop environments? 

This is the only Mac-style global menu setup I've seen that properly replicates the way Mac OS X makes the application name bold to visually separate it from the rest of the menus. Twister OS on Raspberry Pi is a remarkably good macOS imitation, especially after applying Kinto to fix keyboard shortcuts. But it also fails to show the application name in bold letters on the menu. 

I'd like to study the components that install the Macify global menus, to see if I can do the same elsewhere, but I can't quite figure out how to start.

**Comment by jonchun** [comment]
Hi,

I haven't looked at this in a while, but off the top of my head just skimming the source:

you would want these two plasmoids along with latte-dock.

https://github.com/psifidotos/applet-window-title
https://github.com/psifidotos/applet-window-appmenu

From there, you just configure things the way you like them. From a quick search of TwisterOS, it doesn't look like it uses KDE though (understandably... KDE is probably too heavy for a Pi...), so this information likely won't be of too much use to you.

**Comment by RedBearAK** [comment]
Thanks for the links. 

No, Twister OS uses XFCE, which I discovered after needing to do a bit of troubleshooting with the Kinto developer to get Alt+Tab working correctly. We solved some issues with the Kinto installer, but it turned out the main problem was just the existing XFCE keyboard shortcuts interfering. 

Plasmoids are just a KDE thing, I'm assuming. 

After I started this thread I discovered it isn't too difficult to get an application menu in XFCE, including a simple check box to get a bold application title in the appmenu. Not sure how impressed I am with what I ended up with, but it is at least an approximation of the macOS global menu, with indicator icons on the right. It's just the standard XFCE panel moved to the top of the screen with the appmenu plugin added to it and the application buttons removed. 

I think this is the page with the instructions I followed to install the global menu add-on package, called "xfce4-vala-appmenu-plugin": 

https://www.dedoimedo.com/computers/xfce-dock-global-menu.html

This setup in XFCE and a Plank dock (and Albert) is all light enough to use on a very old Pentium 4 single-core laptop with 2GB of RAM. On Linux Mint 19.3, 32-bit. The machine is still dog slow compared to anything even slightly newer, but it is usable to play around with. 

I just realized now that I've figured out how to do this in XFCE that I can probably make the app name bold in Twister OS as well, because they probably use the same plugin to do their global menu. I'm about to try a new Twister OS image with the 1.4.1 update. We'll see what happens.
